### PR TITLE
Added UI tests

### PR DESCRIPTION
**Pull request by krosario23**
## Links to Asana tickets

**Login**
https://app.asana.com/0/1204171546585013/1204501394819402

**Sign-up**
https://app.asana.com/0/1204171546585013/1204501394819406

**Reset Password**
https://app.asana.com/0/1204171546585013/1204501394819408

## Description

Added UI tests for sign-up, login, and reset password components. All three tests succeed when run on my Macbook.

### CODE REVIEW

File: StudySwift/StudySwiftUITests/StudySwiftUITests.swift [hunk 1]
```diff
@@ -22,12 +22,77 @@ final class StudySwiftUITests: XCTestCase {
...
+        
+        let emailField = app.textFields["Email"]
+        XCTAssertTrue(emailField.exists)
+        emailField.tap()
+        emailField.typeText("example@domain.com")
+        
+        let passwordField = app.secureTextFields["Password"]
+        XCTAssertTrue(passwordField.exists)
+        passwordField.tap()
+        passwordField.typeText("password")
```
Comment (MyatThuKo) [inline StudySwift/StudySwiftUITests/StudySwiftUITests.swift:38]: I'd change this password to one that contains special characters and numbers in it. For example, P@ssword12356.

File: StudySwift/StudySwiftUITests/StudySwiftUITests.swift [hunk 2]
```diff
@@ -22,12 +22,77 @@ final class StudySwiftUITests: XCTestCase {
...
+        
+        let confirmField = app.secureTextFields["Confirm Password"]
+        XCTAssertTrue(confirmField.exists)
+        confirmField.tap()
+        confirmField.typeText("password")
```
Comment (MyatThuKo) [inline StudySwift/StudySwiftUITests/StudySwiftUITests.swift:43]: The above comment goes the same here.

**Review by MyatThuKo — CHANGES_REQUESTED** [review]

**Review by MyatThuKo — APPROVED** [review]
Tested and you can merge this PR without doing any changes I suggested but just preference on those changes and not blocking.

**Comment by cyanChill** [comment]
Backend tests actions are failing since the `GOOGLE_SERVICES_JSON` repository secret isn't defined (not sure if it's running off of the actions file in the main repository or on the fork).

**Review by cyanChill — COMMENTED** [review]
Got tests working and they all pass.

**A problem I see is that since your branch currently doesn't contain any of the hardcoded screens since those screens were only pushed from a branch to another branch, they will fail if run normally.**
- **I might suggest pushing this to the branch where the hard coded screens were finalized.**

**Pull request merged**